wal/live_reader.go

```diff
@@ -27,26 +27,40 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+// liveReaderMetrics holds all metrics exposed by the LiveReader.
+type liveReaderMetrics struct {
```

Comment:
Doesn't the struct need to be public, and the metrics private? So that the struct can be passed around in Prometheus?

Reply:
It does not need to be so all good.

However go style guide says it's antipattern (: